universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -99,43 +106,64 @@ public boolean addTranslation(
   }
 
   /**
-   * Validates a new question and checks for path conflicts. This can't be used to validate udpates
-   * because paths will always conflict.
+   * Check for path conflicts. This is to be only used with new questions because updated questions
+   * conflict with themselves, and new versions of questions conflict with previous versions.
    */
-  private ImmutableSet<CiviFormError> validateNewQuestion(QuestionDefinition newDefinition) {
-    ImmutableSet<CiviFormError> errors = newDefinition.validate();
-    if (!errors.isEmpty()) {
-      return errors;
-    }
-    Path newPath = newDefinition.getPath();
+  private ImmutableSet<CiviFormError> validateNewQuestionPath(
+      QuestionDefinition questionDefinition) {
     Optional<Question> maybeConflict =
-        questionRepository.findConflictingQuestion(newPath).toCompletableFuture().join();
+        questionRepository.findPathConflictingQuestion(questionDefinition);
     if (maybeConflict.isPresent()) {
-      Question question = maybeConflict.get();
+      Question conflict = maybeConflict.get();
       return ImmutableSet.of(
           CiviFormError.of(
               String.format(
-                  "path '%s' conflicts with question: %s", newPath.path(), question.getPath())));
+                  "path '%s' conflicts with question id: %s",
+                  questionDefinition.getPath(), conflict.id)));
```

Comment:
Suggestion (non-blocking): I feel path is more readable than an id as this error message is shown to users. Also I am not completely sure integer works with %s.